### PR TITLE
Add code to remove old lighttpd config files left over from v5. 

### PR DESCRIPTION
**Pull request by PromoFaux**
### **What does this PR aim to accomplish?:**

in V6 we will no longer have lighttpd - therefore if we are coming from a v5.x version, we should remove any of lighttpd config we have added to avoid confusion.

Basically reverses what we do here:

https://github.com/pi-hole/pi-hole/blob/6a45c6a8e027e1ac30d4556a88f31684bc80ccf1/automated%20install/basic-install.sh#L1402-L1472

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
- [x] I have read the above and my PR is ready for review. _Check this box to confirm_

### CODE REVIEW

File: automated install/basic-install.sh [hunk 1]
```diff
@@ -1001,6 +1001,38 @@ remove_old_dnsmasq_ftl_configs() {
...
+    local lighttpdConfig="/etc/lighttpd/lighttpd.conf"
+    local condfd="/etc/lighttpd/conf.d/pihole-admin.conf"
+    local confavailable="/etc/lighttpd/conf-available/15-pihole-admin.conf"
+    local confenabled="/etc/lighttpd/conf-enabled/15-pihole-admin.conf"
+
+
+    if [[ -d "/etc/lighttpd/conf.d" ]]; then
+        if grep -q -F 'include "/etc/lighttpd/conf.d/pihole-admin.conf"' "${lighttpdConfig}"; then
+           sed -i '/include "\/etc\/lighttpd\/conf.d\/pihole-admin.conf"/d' "${lighttpdConfig}"
+        fi
```
Comment (rdwebdesign) [inline automated install/basic-install.sh:1014]: Do we need the `if` here?

```suggestion
        sed -i '/include "\/etc\/lighttpd\/conf.d\/pihole-admin.conf"/d' "${lighttpdConfig}"
```

File: automated install/basic-install.sh [hunk 2]
```diff
@@ -1001,6 +1001,38 @@ remove_old_dnsmasq_ftl_configs() {
...
+
+
+    elif [[ -d "/etc/lighttpd/conf-available" ]]; then
+        if is_command lighty-disable-mod ; then
+            lighty-disable-mod pihole-admin > /dev/null || true
+        fi
+
+        if [[ -f "${confavailable}" ]]; then
+            rm "${confavailable}"
+        fi
```
Comment (rdwebdesign) [inline automated install/basic-install.sh:1028]: Disable the config is fine, but I'm not sure if you should remove the _conf-available_ file.

Comment (PromoFaux) [reply automated install/basic-install.sh:1028]: Why not? The file will not be needed in the future, so we should tidy it up after ourselves.

File: automated install/basic-install.sh (hunk 1 above)
Comment (PromoFaux) [reply automated install/basic-install.sh:1014]: Possibly not, I was going for the reverse of what we currently to, so maybe too verbose.

File: automated install/basic-install.sh (hunk 2 above)
Comment (rdwebdesign) [reply automated install/basic-install.sh:1028]: Usually I don't like to touch third party config files, but this one was created by Pi-hole. I guess you're right.

File: automated install/basic-install.sh [hunk 3]
```diff
@@ -1001,6 +1001,36 @@ remove_old_dnsmasq_ftl_configs() {
...
 }
 
+remove_old_pihole_lighttpd_configs() {
+    local lighttpdConfig="/etc/lighttpd/lighttpd.conf"
+    local condfd="/etc/lighttpd/conf.d/pihole-admin.conf"
+    local confavailable="/etc/lighttpd/conf-available/15-pihole-admin.conf"
+    local confenabled="/etc/lighttpd/conf-enabled/15-pihole-admin.conf"
+
+
+    if [[ -d "/etc/lighttpd/conf.d" ]]; then
```
Comment (yubiuser) [inline automated install/basic-install.sh:1011]:
```suggestion
    if [[ -f "${lighttpdConfig}" ]]; then
```

File: automated install/basic-install.sh [hunk 4]
```diff
@@ -1001,6 +1001,36 @@ remove_old_dnsmasq_ftl_configs() {
...
+        sed -i '/include "\/etc\/lighttpd\/conf.d\/pihole-admin.conf"/d' "${lighttpdConfig}"
+
```
Comment (yubiuser) [inline automated install/basic-install.sh:1013]:
```suggestion
fi
```

File: automated install/basic-install.sh [hunk 5]
```diff
@@ -1001,6 +1001,36 @@ remove_old_dnsmasq_ftl_configs() {
...
+        if [[ -f "${condfd}" ]]; then
+            rm "${condfd}"
+        fi
+
+
+    elif [[ -d "/etc/lighttpd/conf-available" ]]; then
```
Comment (yubiuser) [inline automated install/basic-install.sh:1019]:
```suggestion

```

File: automated install/basic-install.sh [hunk 6]
```diff
@@ -1001,6 +1001,36 @@ remove_old_dnsmasq_ftl_configs() {
...
+        fi
+
+        if [[ -f "${confavailable}" ]]; then
+            rm "${confavailable}"
+        fi
+
+        if [[ -f "${confenabled}" ]]; then
+            rm "${confenabled}"
+        fi
+     fi
```
Comment (yubiuser) [inline automated install/basic-install.sh:1031]:
```suggestion

```

**Review by yubiuser — COMMENTED** [review]
As we don't know which crazy user configurations are out there I would not start nested `if...elif` constructs. Just test for the presents of all files individually and remove them if they exist

**Comment by PromoFaux** [comment]
> Just test for the presence of all files individually and remove them if they exist 

Yeah, agreed. Needed to see it all together, was difficult to follow in git suggestions

File: automated install/basic-install.sh [hunk 7]
```diff
@@ -1001,6 +1001,36 @@ remove_old_dnsmasq_ftl_configs() {
...
 }
 
+remove_old_pihole_lighttpd_configs() {
+    local lighttpdConfig="/etc/lighttpd/lighttpd.conf"
+    local condfd="/etc/lighttpd/conf.d/pihole-admin.conf"
+    local confavailable="/etc/lighttpd/conf-available/15-pihole-admin.conf"
+    local confenabled="/etc/lighttpd/conf-enabled/15-pihole-admin.conf"
+
+
+    if [[ -f "${lighttpdConfig}" ]]; then
```
Comment (yubiuser) [inline automated install/basic-install.sh:1011]: You could also remove this outer `if` here and only check for `if grep -q -F 'include "/etc/lighttpd/conf.d/pihole-admin.conf"' "${lighttpdConfig}"; then`

File: automated install/basic-install.sh [hunk 8]
```diff
@@ -1008,27 +1008,27 @@ remove_old_pihole_lighttpd_configs() {
     local confenabled="/etc/lighttpd/conf-enabled/15-pihole-admin.conf"
 
 
-    if [[ -d "/etc/lighttpd/conf.d" ]]; then
-        sed -i '/include "\/etc\/lighttpd\/conf.d\/pihole-admin.conf"/d' "${lighttpdConfig}"
-
-        if [[ -f "${condfd}" ]]; then
-            rm "${condfd}"
+    if [[ -f "${lighttpdConfig}" ]]; then
+        if grep -q -F 'include "/etc/lighttpd/conf.d/pihole-admin.conf"' "${lighttpdConfig}"; then
```
Comment (rdwebdesign) [inline automated install/basic-install.sh:1012]: Why did you add the `if` again?

If the file exists _(line 1011 makes sure it does)_ there's no need to test the existence of the line with `grep`. Just use `sed`.

**Review by rdwebdesign — DISMISSED** [review]

**Comment by yubiuser** [comment]
Was b4d76c37cb9fe5c9046302cd4a9d26c69b2057e3 cherry-picked on purpose? It's part of `development` already - at some point we need to rebase/merge `development` anyway as we're going to keep the branch around (and allow for changes +  Dependabot).

**Review by yubiuser — APPROVED** [review]

**Comment by OfficialMuffin** [comment]
Does this mean PiHole will not be using lighttpd anymore?

**Comment by PromoFaux** [comment]
Correct... pihole-FTL now ships with a built in webserver

**Comment by OfficialMuffin** [comment]
Does that mean I no longer need lighttpd? Is it safe to remove the package for my system?

22 Feb 2025, 20:59 by ***@***.***:

**Comment by jfb-pihole** [comment]
You  no longer need lighttpd for Pi-hole.  If it is providing other services to you, then you would want to keep it (other web pages, etc.).

If not needed, you can safely uninstall it with no impact on Pi-hole.

**Comment by yubiuser** [comment]
Please create a new issue and provide a debug token.
